feat: implement disk-based CI logging with SSH transfer

ludamad opened the pull request:
## Summary

Implement disk-based CI logging system that writes logs to both Redis and persistent disk storage on bastion, solving the problem of 120GB+ compressed logs overwhelming Redis (which has only 2-week retention).

## Changes

### Infrastructure
- 1TB EBS volume mounted at `/logs-disk` on CI bastion in us-east-2

### Core Implementation
- **`ci3/source_cache`**: New module with `cache_persistent()` function for dual Redis+disk writes
- **`ci3/cache_log`**, **`ci3/run_test_cmd`**, **`ci3/denoise`**: Updated to use `cache_persistent()` for final log writes
- **`ci3/bootstrap_ec2`**: SSH key passing via base64 encoding (following GCP_SA_KEY pattern)
- **`ci3/aws/build_instance_ssh_config`**: SSH multiplexing config for efficient persistent connections
- **`rkapp/rk.py`**: In [iac](https://github.com/AztecProtocol/iac/pull/18). Disk fallback when Redis key expires

### Architecture
- **Write path**: Live updates (every 5s) → Redis only. Final write → Redis + SSH to bastion disk
- **Read path**: Try Redis first, fall back to `/logs-disk` on bastion if expired
- **SSH transfer**: Background SSH pipe using multiplexed connections (ControlMaster, ControlPersist 10m)
- **Efficiency**: Single long-lived SSH connection shared across all log writes (zero latency after first connect)

johnathan79717 commented on ci3/denoise:
```diff
@@ -71,6 +72,23 @@ function publish_log {
+    cat $outfile
+  } | cache_persistent $key $CI_REDIS_EXPIRE
+}
+
```
Nit: Maybe we can avoid the repetition with something like:

```suggestion
function _format_log_output {
  echo "Parent Log: ${PARENT_LOG_URL:-none}"
  echo "Command: $cmd"
  echo "Date: $(date)"
  echo "Status: $status"
  echo "Took: ${time}"
  echo
  cat $outfile
}

function publish_log {
  [ "$CI_REDIS_AVAILABLE" -eq 0 ] && return
  _format_log_output | redis_setexz $key $CI_REDIS_EXPIRE
}

function publish_log_final {
  [ "$CI_REDIS_AVAILABLE" -eq 0 ] && return
  _format_log_output | cache_persistent $key $CI_REDIS_EXPIRE
}

```

ludamad committed:
feat: implement disk-based CI logging with SSH transfer

## Summary

Implement disk-based CI logging system that writes logs to both Redis and persistent disk storage on bastion, solving the problem of 120GB+ compressed logs overwhelming Redis (which has only 2-week retention).

## Changes

### Infrastructure
- 1TB EBS volume mounted at `/logs-disk` on CI bastion in us-east-2

### Core Implementation
- **`ci3/source_cache`**: New module with `cache_persistent()` function for dual Redis+disk writes
- **`ci3/cache_log`**, **`ci3/run_test_cmd`**, **`ci3/denoise`**: Updated to use `cache_persistent()` for final log writes
- **`ci3/bootstrap_ec2`**: SSH key passing via base64 encoding (following GCP_SA_KEY pattern)
- **`ci3/aws/build_instance_ssh_config`**: SSH multiplexing config for efficient persistent connections
- **`rkapp/rk.py`**: In [iac](https://github.com/AztecProtocol/iac/pull/18). Disk fallback when Redis key expires

### Architecture
- **Write path**: Live updates (every 5s) → Redis only. Final write → Redis + SSH to bastion disk
- **Read path**: Try Redis first, fall back to `/logs-disk` on bastion if expired
- **SSH transfer**: Background SSH pipe using multiplexed connections (ControlMaster, ControlPersist 10m)
- **Efficiency**: Single long-lived SSH connection shared across all log writes (zero latency after first connect)
